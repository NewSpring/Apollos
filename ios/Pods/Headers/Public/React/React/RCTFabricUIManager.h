../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h