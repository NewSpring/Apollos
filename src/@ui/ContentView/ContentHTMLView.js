import { withProps } from 'recompose';
import HTMLView, { defaultRenderer } from '@ui/HTMLView';

const renderer = (node, { children, ...other }) => { // eslint-disable-line
  // the defaultRenderer support several basic elements out of the box,
  // this function only needs to handle the cases that are unique to content.
  return defaultRenderer(node, { children, ...other });
};

const ContentHTMLView = withProps({
  renderer,
})(HTMLView);

export default ContentHTMLView;
