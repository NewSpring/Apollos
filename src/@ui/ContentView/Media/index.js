export { default as ImageHeader } from './ImageHeader';
export { default as VideoHeader } from './VideoHeader';
export { default as AudioBanner } from './AudioBanner';
