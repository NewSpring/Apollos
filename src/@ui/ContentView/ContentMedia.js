import React from 'react';
import { View, Platform, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import Color from 'color';

import ConnectedImage from '@ui/ConnectedImage';
import EmbeddedVideoPlayer from '@ui/EmbeddedVideoPlayer';
import styled from '@ui/styled';
import LinearGradient from '@ui/LinearGradient';

const ImageOverlay = styled({
  ...StyleSheet.absoluteFillObject,
})(LinearGradient);

const getGradientValues = (overlayColor) => {
  const values = {
    colors: [`${Color(overlayColor).fade(1).string()}`, overlayColor],
    start: [0, 0],
    end: [0, 1],
    locations: [0.3, 1],
  };

  return values;
};

const ImageHeader = styled({
  width: '100%',
  aspectRatio: 1,
  resizeMode: 'cover',
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '50%',
    },
  }),
})(ConnectedImage);

const VideoHeader = styled({
  width: '100%',
  aspectRatio: 16 / 9,
  ...Platform.select({
    web: StyleSheet.absoluteFillObject,
  }),
})(EmbeddedVideoPlayer);

const VideoWrapper = styled(Platform.select({
  web: {
    position: 'relative',
    paddingBottom: '56.25%', // NOTE: 16:9 :D
    overflow: 'hidden',
    width: '100%',
    height: 0,
  },
}))(View);

const ContentMedia = ({
  video,
  images = [],
  imageOverlayColor,
}) => {
  let visual = null;
  if (video && video.embedUrl) {
    visual = (
      <VideoWrapper>
        <VideoHeader src={video.embedUrl} />
      </VideoWrapper>
    );
  } else if (images && images.length) {
    visual = (
      <View>
        <ImageHeader source={images} />
        {imageOverlayColor && (
          <ImageOverlay
            colors={getGradientValues(imageOverlayColor).colors}
            start={getGradientValues(imageOverlayColor).start}
            end={getGradientValues(imageOverlayColor).end}
            locations={getGradientValues(imageOverlayColor).locations}
          />
        )}
      </View>
    );
  }

  // TODO: later this should support audio button below player
  return visual;
};

ContentMedia.propTypes = {
  video: PropTypes.shape({
    embedUrl: PropTypes.string,
  }),
  images: PropTypes.any, // eslint-disable-line
  imageOverlayColor: PropTypes.string,
};

export default ContentMedia;
