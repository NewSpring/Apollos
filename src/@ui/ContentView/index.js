import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import styled from '@ui/styled';
import Placeholder from '@ui/Placeholder';

import { VideoHeader, ImageHeader, AudioBanner } from './Media';

export { default as HTMLView } from '@ui/HTMLView';
export { H3 as Title, H4 as SubHeading } from '@ui/typography';
export { default as ByLine } from './ByLine';

const enhance = compose(
  pure,
  setPropTypes({
    contentId: PropTypes.string,
    authors: PropTypes.arrayOf(PropTypes.string),
    title: PropTypes.string,
    subheading: PropTypes.string,
    body: PropTypes.string,
    video: PropTypes.shape({
      embedUrl: PropTypes.string,
    }),
    audio: PropTypes.arrayOf(
      PropTypes.shape({
        duration: PropTypes.string,
        file: PropTypes.string,
      }),
    ),
    seriesImages: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
      }),
    ),
    seriesColors: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.string })),
    images: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
      }),
    ),
    thumbnail: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
      }),
    ),
    imageOverlayColor: PropTypes.string,
  }),
);

const ContentWrapper = styled(({ theme }) => ({
  padding: theme.sizing.baseUnit,
}))(View);

const renderHeader = ({
  video, images = [], thumbnailImage, imageOverlayColor,
}) => {
  let headerType = null;
  if (video && video.embedUrl) {
    headerType = <VideoHeader source={video} />;
  } else if (images && images.length) {
    headerType = (
      <ImageHeader
        images={images}
        thumbnail={thumbnailImage}
        imageOverlayColor={imageOverlayColor}
      />
    );
  }

  return headerType;
};

const renderAudioBar = ({
  contentId, title, audio, seriesImages, seriesColors,
}) => {
  let audioComponent = null;
  if (audio && audio.length) {
    const track = {
      title,
      ...audio[0],
    };

    audioComponent = (
      <AudioBanner
        mediaId={contentId}
        currentTrack={track}
        playlist={{
          title,
          images: seriesImages,
          colors: seriesColors,
          tracks: [track],
        }}
      />
    );
  }

  return audioComponent;
};

const ContentView = enhance(
  ({
    contentId,
    video,
    images = [],
    thumbnailImage,
    seriesImages = [],
    seriesColors = [],
    imageOverlayColor = '',
    title,
    audio,
    children,
    isLoading,
  }) => (
    <View>
      {renderHeader({
        video, images, thumbnailImage, imageOverlayColor,
      })}
      {renderAudioBar({
        contentId, title, audio, seriesImages, seriesColors,
      })}
      <ContentWrapper>
        {children}
        {isLoading ? <Placeholder.Paragraph /> : null}
      </ContentWrapper>
    </View>
  ),
);

export default ContentView;
