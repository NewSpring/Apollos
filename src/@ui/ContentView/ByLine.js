import React from 'react';
import PropTypes from 'prop-types';
import { startCase } from 'lodash';
import { compose, pure, branch, renderNothing } from 'recompose';
import { H6 } from '@ui/typography';

const enhance = compose(
  pure,
  branch(({ authors }) => !authors || !authors.length, renderNothing),
);

const ByLine = enhance(({ authors }) => (
  <H6>By: {startCase(authors.join(', '))}</H6>
));

ByLine.propTypes = {
  authors: PropTypes.arrayOf(PropTypes.string),
};

export default ByLine;
