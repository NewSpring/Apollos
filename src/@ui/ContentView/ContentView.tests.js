import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '@ui/TestProviders';
import ContentView, { ByLine, Title, HTMLView } from './';

describe('the ContentView component', () => {
  it('renders', () => {
    const tree = renderer.create(
      <Providers>
        <ContentView>
          <Title>The great escape</Title>
          <ByLine authors={['Mike Douglas']} />
          <HTMLView>{'<p>Yo yo</p><p>this is some cool html</p>'}</HTMLView>
        </ContentView>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
