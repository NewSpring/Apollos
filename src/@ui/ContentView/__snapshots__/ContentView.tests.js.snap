// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ContentView component renders 1`] = `
<View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 32.4,
          "lineHeight": 41.04,
          "paddingBottom": 2.7,
          "paddingTop": 5.4,
        }
      }
    >
      The great escape
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 15.75,
          "lineHeight": 17.75,
          "paddingBottom": 9,
          "paddingTop": 4.5,
        }
      }
    >
      By: 
      Mike Douglas
    </Text>
    <View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Yo yo
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            this is some cool html
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ContentView component renders with images 1`] = `
<View>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#dddddd",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Image
            collapsable={undefined}
            onLoad={[Function]}
            source={
              Array [
                Object {
                  "height": 400,
                  "uri": "https://picsum.photos/600/400/?random",
                  "width": 600,
                },
                Object {
                  "height": 400,
                  "uri": "https://picsum.photos/600/400/?random",
                  "width": 600,
                },
              ]
            }
            style={
              Object {
                "backgroundColor": "#dddddd",
                "height": "100%",
                "opacity": 0,
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
      <ExponentLinearGradient
        colors={
          Array [
            16416882,
            4294606962,
          ]
        }
        endPoint={
          Array [
            0,
            1,
          ]
        }
        locations={
          Array [
            0.3,
            1,
          ]
        }
        startPoint={
          Array [
            0,
            0,
          ]
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 32.4,
          "lineHeight": 41.04,
          "paddingBottom": 2.7,
          "paddingTop": 5.4,
        }
      }
    >
      The great escape
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 15.75,
          "lineHeight": 17.75,
          "paddingBottom": 9,
          "paddingTop": 4.5,
        }
      }
    >
      By: 
      Mike Douglas
    </Text>
    <View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Yo yo
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            this is some cool html
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ContentView component renders with video 1`] = `
<View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 32.4,
          "lineHeight": 41.04,
          "paddingBottom": 2.7,
          "paddingTop": 5.4,
        }
      }
    >
      The great escape
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 15.75,
          "lineHeight": 17.75,
          "paddingBottom": 9,
          "paddingTop": 4.5,
        }
      }
    >
      By: 
      Mike Douglas
    </Text>
    <View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Yo yo
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            this is some cool html
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
